mautic/mautic
# Form action - Download latest asset from category

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds new field to the **Download Asset** form action - Category:

![edit form latest asset download mautic](https://user-images.githubusercontent.com/1235442/42698909-8c92849a-86bf-11e8-8f00-f49591e09446.png)

The Asset select box has new "empty" option **Use the latest asset from the category**. If this empty option is selected then the action will take the latest asset which belongs to the selected category.

This is useful if you're adding often new versions of the file that is suppose to be downloadable after form submission. With this configuration you won't have to upload the asset and change the form action. Adding the new asset will do it.

#### Steps to test this PR:
1. Make sure the download actions still work on some old forms.
2. Create a new category.
3. Create an asset and add it to the category.
4. Create a form with the Download Asset action.
5. Select the category.
6. Save the form.
7. Go to the preview page.
8. Submit the form.
9. You should get the asset from point 3. even though it's not directly specified in the form action.
10. Add new asset to the category.
11. Now the form submission will serve the new asset.
12. Delete both assets.
13. Submit the form.
14. No asset is served, the form submission was saved correctly though.
